### PR TITLE
Add tests

### LINKED ISSUE #3: Add basic CI
It is very important to have some basic CI here.

Some good starting points:

- Test the template is produced OK with default input.
- Test the github actions syntax.

### PR DESCRIPTION
**Pull request by joao-p-marques**
Fixes #3

@Tecnativa
TT27819

ping @Yajo

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - master
```
Comment (yajo) [inline .github/workflows/test.yml]:
```suggestion
```

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import pytest
+import yaml
+from plumbum import local
+from plumbum.cmd import git
+
+with open("copier.yml") as copier_fd:
+    COPIER_SETTINGS = yaml.safe_load(copier_fd)
+
+
+@pytest.fixture()
```
Comment (yajo) [inline tests/conftest.py]: Doing this once per session should be enough, right?

```suggestion
@pytest.fixture(scope="session")
```

File: tests/github-workflow.json [hunk 3]
```diff
@@ -0,0 +1,1181 @@
+{
```
Comment (yajo) [inline tests/github-workflow.json]: This file comes from https://github.com/SchemaStore/schemastore/blob/15b1babd951184784e608feade64620235f99485/src/schemas/json/github-workflow.json, right?

Delete it and use a git submodule then.

File: tests/test_default_settings.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    schema_url = "https://json.schemastore.org/github-workflow.json"
+    schema_path = "github-workflow.json"
+    # Try to fetch directly from server to get latest version
+    try:
+        logger.info(f"Trying to fetch data from {schema_url}")
+        data = requests.get(schema_url).content
+    except requests.exceptions.RequestException:
+        logger.info(f"Fetching from URL failed, using file {schema_path}")
+        with open(schema_path, "r") as file:
+            data = file.read()
```
Comment (yajo) [inline tests/test_default_settings.py]: Use the git submodule version only, always.

File: tests/test_default_settings.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+def validate_schema(yaml_data):
+    """Validate yaml based on schema."""
+    expected_schema = get_schema()
+    try:
+        jsonschema.validate(instance=yaml_data, schema=expected_schema)
+    except jsonschema.exceptions.ValidationError as err:
+        logger.error(err)
+        return False
+    return True
```
Comment (yajo) [inline tests/test_default_settings.py]: I don't see much sense on slurping the exception to print the error manually... just let the exception raise and pytest will print the full error stack.

```suggestion
    jsonschema.validate(instance=yaml_data, schema=expected_schema)
```

File: tests/test_default_settings.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                "\non:", "\n'on':"
+            )  # HACK: pyyaml interprets "on" key as "True", causing a false negative on the validation function
+            yaml_data = yaml.safe_load(processed_content)
+            # Ensure project data propagated
+            assert (
+                yaml_data["jobs"]["build-push"]["env"]["DOCKERHUB_IMAGE_NAME"]
+                == "test/test"
+            )
+            # Validate according to Github Actions expected syntax
+            assert validate_schema(yaml_data)
```
Comment (yajo) [inline tests/test_default_settings.py]:
```suggestion
            validate_schema(yaml_data)
```

File: tests/test_default_settings.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        # Tests shouldn't exist
+        assert not Path("pytest.ini").exists()
+        assert not Path("pyproject.toml").exists()
+        assert not Path("tests/conftest.py").exists()
+        # Validate CI config
+        with Path(".github", "workflows", "ci.yml").open("r") as f:
+            content = f.read()
+            processed_content = content.replace(
+                "\non:", "\n'on':"
+            )  # HACK: pyyaml interprets "on" key as "True", causing a false negative on the validation function
```
Comment (yajo) [inline tests/test_default_settings.py]: I think it makes more sense to just add quotes around those "on" in the original file, IMHO.

File: tests/test_default_settings.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+            yaml_data = yaml.safe_load(processed_content)
+            # Validate according to Github Actions expected syntax
+            assert validate_schema(yaml_data)
```
Comment (yajo) [inline tests/test_default_settings.py]:
```suggestion
            validate_schema(yaml_data)
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by joao-p-marques** [comment]
Thanks for the review @Yajo!

I've made the changes. I'm not sure if the transition to the git submodule is completely correct, as I had to make some changes in the tests with the passing of `cloned_template` between the functions. Is there a better way to do that?

File: tests/test_default_settings.py [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    schema_file = Path(
+        cloned_template,
+        "tests",
+        "schemastore",
+        "src",
+        "schemas",
+        "json",
+        "github-workflow.json",
+    )
+    logger.info(f"Fetching schema from {schema_file}")
```
Comment (yajo) [inline tests/test_default_settings.py]: I don't think this log is valuable anymore. Besides, you already "fetched" it.

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Your usage of submodules seems good.

**Comment by joao-p-marques** [comment]
Thanks! Merging then :+1:

**Pull request merged**